### DISCUSSION TITLE
Â appearing before £

### DISCUSSION THREAD

**Discussion by CustomBrandCo** [cefsharp/CefSharp#4626] [General]
### Is there an existing issue for this?

- [X] I have searched both open/closed issues, no issue already exists.

### CefSharp Version

119.1.20

### Operating System

Windows 11

### Architecture

x64

### .Net Version

6.0

### Implementation

WPF

### Reproduction Steps

[test.txt](https://github.com/cefsharp/CefSharp/files/13354214/test.txt)

I have attached html that I am displaying in WPF. If I save this as a html file and open in Chrome or Edge, it displays fine.

However, in WPF, it looks like this:

![image](https://github.com/cefsharp/CefSharp/assets/143000246/1f7b00eb-49cc-4500-8ccc-1ba39ec2a265)

### Expected behavior

Â to not appear before £ signs

### Actual behavior

Â appearing before £ signs

### Does this problem also occur in the CEF Sample Application

Not Tested

**Comment by amaitland** [comment]
> I have attached html that I am displaying in WPF. If I save this as a html file and open in Chrome or Edge, it displays fine.
> 
> However, in WPF, it looks like this:

How do you reproduce the problem? There are many different ways to load a file. Please provide a clear list of steps.

How are you opening the file in `Chrome`?

Does your file have a BOM? https://en.wikipedia.org/wiki/Byte_order_mark

Try adding one if it doesn't already have one.

> ### Does this problem also occur in the CEF Sample Application
> Not Tested

Is there a reason you skipped this testing?

**Reply by CustomBrandCo** [reply]
I have tried loading the file in WPF using the following code:

`var html = System.IO.File.ReadAllText(@"************\test.txt");
ChromiumWebBrowser.LoadHtml(html);`

Where ChromiumWebBrowser is an instance of the control in a window.

I am also trying to load a html string from SQL, using entity framework, but it gives me the same result.

The text is UTF-8 encoded and from what I understand, having a BOM in UTF-8 is not recommended.

I am opening the file in Chrome or Edge by changing the extension from .txt to .html. I would have uploaded the file as a html doc, but GitHub doesn't allow it.

**Reply by amaitland** [reply]
> The text is UTF-8 encoded and from what I understand, having a BOM in UTF-8 is not recommended.

`Chromium` supports BOM, there are cases where it's very useful. You've read the file in `.Net` yourself, so this isn't one of those cases.

> var html = System.IO.File.ReadAllText(@"************\test.txt"); ChromiumWebBrowser.LoadHtml(html);

Your not comparing like for like when you open the file in `Chrome/Edge` directly (assuming you are using the `file` scheme).

You've loaded html as string, without specifying a URL it's converted to a base64 encoded data uri. There's no encoding specified by default.

```c#
//Option 1
var html = System.IO.File.ReadAllText(@"************\test.txt");
ChromiumWebBrowser.LoadHtml(html, "http://webpage.test");

//Option 2
var htmlString = new HtmlString(html, true, "UTF-8");

browser.LoadUrl(htmlString.ToDataUriString());
```

`Option 1` will load the html with a url using a [ResourceHandler](https://github.com/cefsharp/CefSharp/wiki/General-Usage#resourcehandler)

I'd encourage you to review the source for `LoadHtml` to get a better understanding of what's going on.
